### PR TITLE
rainbow tabs

### PR DESCRIPTION
**Pull request by dereknelson**
rainbow's navigation hit up the dark web & ingested 300 micrograms of that react-navigation v6 in hopes its users will discover a new place to discover themselves on our discover page

## What changed (plus any additional context for devs)
hopefully nothing changed besides having tabs on the bottom

## Screen recordings / screenshots
RocketSim_Screenshot_iPhone_14_Pro_2023-09-15_11 58 02

## What to test
navigate around & make sure no corner of the app is unnavigable

### CODE REVIEW

**Comment by dereknelson** [comment]
officially rdy for review!

File: src/components/asset-list/RecyclerAssetList2/profile-header/ProfileActionButtonsRow.tsx [hunk 1]
```diff
@@ -274,70 +273,18 @@ function MoreButton() {
...
-    </ContextMenuButton>
+    <>
+      {/* @ts-expect-error JavaScript component */}
+      <CopyFloatingEmojis textToCopy={accountAddress}>
+        <ActionButton
+          onPress={handlePressCopy}
+          icon="􀐅"
+          testID="receive-button"
+        >
+          {'Copy'}
```
Comment (benisgold) [inline src/components/asset-list/RecyclerAssetList2/profile-header/ProfileActionButtonsRow.tsx:285]: i18n

File: src/components/qrcode-scanner/QRCodeScanner.tsx [hunk 2]
```diff
@@ -105,9 +110,24 @@ export default function QRCodeScanner() {
     }
   }, []);
 
-  useEffect(() => {
+  useFocusEffect(() => {
+    console.log('triggered');
```
Comment (benisgold) [inline src/components/qrcode-scanner/QRCodeScanner.tsx:114]: delete

File: src/hooks/useImportingWallet.ts [hunk 3]
```diff
@@ -274,6 +275,7 @@ export default function useImportingWallet({ showImportModal = true } = {}) {
           ? resolvedAddress
           : sanitizeSeedPhrase(seedPhrase);
 
+        console.log('!!!!!!!!!!! 1/2');
```
Comment (benisgold) [inline src/hooks/useImportingWallet.ts:278]: sir

File: src/hooks/useImportingWallet.ts [hunk 4]
```diff
@@ -286,10 +288,12 @@ export default function useImportingWallet({ showImportModal = true } = {}) {
             image,
             true
           );
+          console.log('!!!!!!!!!!! 1');
```
Comment (benisgold) [inline src/hooks/useImportingWallet.ts:291]: this

File: src/hooks/useImportingWallet.ts [hunk 5]
```diff
@@ -286,10 +288,12 @@ export default function useImportingWallet({ showImportModal = true } = {}) {
...
           await dispatch(walletsLoadState(profilesEnabled));
           handleSetImporting(false);
         } else {
           const previousWalletCount = keys(wallets).length;
+          console.log('!!!!!!!!!!! 2');
```
Comment (benisgold) [inline src/hooks/useImportingWallet.ts:296]: is

File: src/hooks/useImportingWallet.ts [hunk 6]
```diff
@@ -302,19 +306,25 @@ export default function useImportingWallet({ showImportModal = true } = {}) {
             image
           )
             .then(success => {
+              console.log('!!!!!!!!!!! 3');
```
Comment (benisgold) [inline src/hooks/useImportingWallet.ts:309]: a

File: src/screens/QRScannerScreen.tsx [hunk 7]
```diff
@@ -38,41 +51,142 @@ const ScannerHeader = styled(Header).attrs({
...
-            {android && (
+          <ScannerContainer>
+            <Background />
+            <CameraDimmer cameraVisible={true}>
+              {android && (
+                <ScannerHeader>
+                  <EmulatorPasteUriButton />
+                </ScannerHeader>
+              )}
+              {true && (
```
Comment (benisgold) [inline src/screens/QRScannerScreen.tsx:161]: why this condition???

**Review by benisgold — APPROVED** [review]
looks 🔥 

just some small cleanup comments

File: src/screens/QRScannerScreen.tsx (hunk 7 above)
Comment (dereknelson) [reply src/screens/QRScannerScreen.tsx:161]: @skylarbarrera assuming this is good to remove?
image

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
APP-754 has been fixed.

Found some issues with Android only, and then a ledger issue on both OS's 
all on the bug epic, under sub-issues = https://linear.app/rainbow/issue/APP-802/rainbow-tabs-bug-epic

File: src/components/animations/ButtonPressAnimation/ButtonPressAnimation.android.tsx [hunk 8]
```diff
@@ -217,10 +219,12 @@ const SimpleScaleButton = ({
   // we won't guess if there are any animated styles in there but we can
   // not render the Animated.View if we don't use that prop at all
   const Wrapper = contentContainerStyle ? Animated.View : View;
-
+  // hack to lay out BottomTab buttons normally instead of being stacked on top of each other
+  const [overflowMarginHelper, setOverflowMargin] = useState(overflowMargin);
+  const layoutHelper = () => setOverflowMargin(overflowMargin - 1);
```
Comment (skylarbarrera) [inline src/components/animations/ButtonPressAnimation/ButtonPressAnimation.android.tsx:224]: 🙈

File: src/components/asset-list/RecyclerAssetList/index.tsx [hunk 9]
```diff
@@ -677,6 +678,8 @@ function RecyclerAssetList({
     [isCoinListEdited, setIsCoinListEdited]
   );
 
+  console.log('rendering RecyclerAssetList');
```
Comment (skylarbarrera) [inline src/components/asset-list/RecyclerAssetList/index.tsx:681]: log

File: src/components/asset-list/RecyclerAssetList2/core/useMemoBriefSectionData.ts [hunk 10]
```diff
@@ -95,8 +95,7 @@ export default function useMemoBriefSectionData({
 
         if (
           data.type === CellType.PROFILE_STICKY_HEADER ||
-          data.type === CellType.NFTS_HEADER ||
-          data.type === CellType.POSITIONS_HEADER
+          data.type === CellType.NFTS_HEADER
```
Comment (skylarbarrera) [inline src/components/asset-list/RecyclerAssetList2/core/useMemoBriefSectionData.ts:98]: havnt tested so may be handled somewhere else but I think we can get ride of the NFT_HEADER being sticky as well

File: src/navigation/SwipeNavigator.js [hunk 11]
```diff
@@ -1,43 +1,387 @@
...
+  useEffect(() => {
+    if (!isTap) {
+      animateTabs(state.index, true);
+    } else {
+      animateTabs(state.index);
+    }
+    setIsTap(false);
+  }, [state.index, isTap]);
+
+  // for when QRScannerScreen is re-added
```
Comment (skylarbarrera) [inline src/navigation/SwipeNavigator.js:121]: bumping to make sure Christian is aware of this

File: src/screens/QRScannerScreen.tsx (hunk 7 above)
Comment (skylarbarrera) [reply src/screens/QRScannerScreen.tsx:161]: yeah just revert this change

File: src/screens/QRScannerScreen.tsx [hunk 12]
```diff
@@ -38,41 +51,142 @@ const ScannerHeader = styled(Header).attrs({
...
-        <ScannerContainer>
-          <Background />
-          <CameraDimmer cameraVisible={true}>
-            {android && (
+          <ScannerContainer>
+            <Background />
+            <CameraDimmer cameraVisible={true}>
+              {android && (
+                <ScannerHeader>
+                  <EmulatorPasteUriButton />
```
Comment (skylarbarrera) [inline src/screens/QRScannerScreen.tsx:158]: this shouldnt be only android p sure

File: src/walletConnect/sheets/AuthRequest.tsx [hunk 13]
```diff
@@ -152,6 +152,7 @@ export function AuthRequest({
         <Box paddingBottom="36px">
           <ButtonPressAnimation
             onPress={() => {
+              console.log('here');
```
Comment (skylarbarrera) [inline src/walletConnect/sheets/AuthRequest.tsx:155]: loggr

**Review by skylarbarrera — APPROVED** [review]
lFG

**Review by ibrahimtaveras00 — APPROVED** [review]
All issues have been resolved, QA Passed 👍🏽

File: src/navigation/SwipeNavigator.js (hunk 11 above)
Comment (dereknelson) [reply src/navigation/SwipeNavigator.js:121]: ya we agreed modal nav instead of 4th tab is fine for now

**Review by jinchung — CHANGES_REQUESTED** [review]
Please double check the import wallet flow for watched wallets - looks like it navigates back out to the welcome screen

**Comment by jinchung** [comment]
@benisgold think your latest force push didn't resolve a conflict in DiscoverHome. I just pushed a fix to the homescreen e2e tests, so make sure you pull first

**Comment by dereknelson** [comment]
> Please double check the import wallet flow for watched wallets - looks like it navigates back out to the welcome screen

i watched an address & not seeing this, this is the flow you're referring to correct?

https://github.com/rainbow-me/rainbow/assets/6868432/a2c62850-540f-48a7-af2c-086a4ab10ed7

**Review by jinchung — APPROVED** [review]

**Pull request merged**